src/bartiq/analysis/rewriters/utils.py

```diff
@@ -77,10 +77,13 @@ class Assumption(Instruction):
     symbol_name: str
     comparator: Comparators | str
     value: int | float
-    symbol_properties: dict[str, bool | None] = field(init=False)
 
-    def __post_init__(self):
-        object.__setattr__(self, "symbol_properties", _get_assumption_properties(self.comparator, self.value))
+    @property
+    def symbol_properties(self) -> dict[str, bool | None]:
+        """A dictionary of properties (positive, negative, non-positive, non-negative).
+
+        If None, the truthiness of the property could not be determined."""
```

Comment:
None -> `None` .
Also, perhaps rather than `truthiness` we could say "the boolean value of the property ..." ?